core/include/kernel/interrupt.h

```diff
@@ -30,7 +30,28 @@
 #include <types_ext.h>
 #include <sys/queue.h>
 
+#define ITRF_TRIGGER_MASK	0x1
+#define ITRF_TRIGGER_SHIFT	0x1
 #define ITRF_TRIGGER_LEVEL	(1 << 0)
+#define ITRF_TRIGGER_EDGE	(0 << 0)
+
+#define ITRF_SHARE_MASK		0x2
+#define ITRF_SHARE		(1 << 1)
+#define ITRF_NOT_SHARE		(0 << 1)
+
+#define cpumask_of(cpu)		(1 << cpu)
+
+#define GIC_OK			0
+#define GIC_BAD_PARAMETER	1
+#define GIC_BAD_MEMORY		2
+
+enum itr_return {
+	ITRR_NONE,
+	ITRR_HANDLED,
+};
+
+typedef uint32_t itr_result;
```

Comment:
do we really need a typedef for that? would be much simpler to use a `int` type.